presto-memory/src/main/java/com/facebook/presto/plugin/memory/MemoryMetadata.java

```diff
@@ -102,6 +104,20 @@ public synchronized void createSchema(ConnectorSession session, String schemaNam
         schemas.add(schemaName);
     }
 
+    @Override
+    public synchronized void dropSchema(ConnectorSession session, String schemaName)
+    {
+        if (!schemas.contains(schemaName)) {
+            throw new PrestoException(NOT_FOUND, format("Schema [%s] does not exist", schemaName));
+        }
+
+        tables.values().stream()
+                .filter(table -> table.getSchemaName().equals(schemaName))
+                .forEach(table -> dropTable(session, table));
```

Comment:
Should it be allowed to drop schema with tables?

Reply:
Good point. It does not look as safe thing to do. Thanks.